### PR TITLE
Add it.unibo.tuprolog.ide

### PR DESCRIPTION
**Pull request by VaiTon**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  [tuProlog](https://www.cs.nmsu.edu/ALP/2013/10/tuprolog-making-prolog-ubiquitous/) (2P henceforth) is a multi-paradigm logic programming framework written in Java. The packaged application is an IDE for using Prolog

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission.
  - https://github.com/tuProlog/2p-kt/issues/723
  
  
### Blocked on
- https://github.com/tuProlog/2p-kt/pull/725

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: it.unibo.tuprolog.ide.desktop
Comment (bbhtt) [inline it.unibo.tuprolog.ide.desktop:1]: icon should be upstream https://github.com/tuProlog/2p-kt/

File: it.unibo.tuprolog.ide.png
Comment (bbhtt) [inline it.unibo.tuprolog.ide.png:1]: desktop file should be upstream https://github.com/tuProlog/2p-kt/

File: it.unibo.tuprolog.ide.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+sdk-extensions:
+  - "org.freedesktop.Sdk.Extension.openjdk17"
+command: tuprolog-ide
+finish-args:
+  - "--env=PATH=/app/jre/bin:/app/bin:/usr/bin"
+  - "--env=JAVA_HOME=/app/jre"
+  - "--share=ipc"
+  - "--socket=x11"
+  - "--device=dri"
+  - "--filesystem=home"
```
Comment (bbhtt) [inline it.unibo.tuprolog.ide.yml:14]: who does it need home ?

File: it.unibo.tuprolog.ide.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+id: it.unibo.tuprolog.ide
```
Comment (bbhtt) [inline it.unibo.tuprolog.ide.yml:1]: does upstream have control over tuprolog.unibo.it ? It seems to redirect to somewhere else.

Comment (VaiTon) [reply it.unibo.tuprolog.ide.yml:1]: Yup, it's just a redirect to the wiki used to document the project

File: it.unibo.tuprolog.ide.yml (hunk 1 above)
Comment (VaiTon) [reply it.unibo.tuprolog.ide.yml:14]: I tried without home and I was not able to save files outside the container. Maybe I'm missing something here. Any way to do it better?

File: it.unibo.tuprolog.ide.png
Comment (VaiTon) [reply it.unibo.tuprolog.ide.png:1]: The file is at [1]. Should I download it during the build process?

[1]: https://github.com/tuProlog/2p-kt/tree/master/.img

Comment (bbhtt) [reply it.unibo.tuprolog.ide.png:1]: add it as a source like `type: file`, `url:` and `sha256`, same for icon. Pull the file from a git commit not a branch in the link.

File: it.unibo.tuprolog.ide.yml (hunk 1 above)
Comment (bbhtt) [reply it.unibo.tuprolog.ide.yml:14]: i guess it doesn't support the filechooser portal.

File: it.unibo.tuprolog.ide.desktop
Comment (VaiTon) [reply it.unibo.tuprolog.ide.desktop:1]: I opened an issue about this: https://github.com/tuProlog/2p-kt/issues/724

Is this blocking?

File: it.unibo.tuprolog.ide.yml (hunk 1 above)
Comment (VaiTon) [reply it.unibo.tuprolog.ide.yml:14]: I don't think it does, unfortunately. I could limit to a specific folder like Documents though, but I don't know

**Comment by bbhtt** [comment]
bot, build it.unibo.tuprolog.ide

**Comment by bbhtt** [comment]
You aren't installing any metainfo file. It's in the requirements page linked in PR checklist.

**Comment by VaiTon** [comment]
bot, build it.unibo.tuprolog.ide

File: it.unibo.tuprolog.ide.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        dest-filename: tuprolog-ide.jar
+      - type: file
+        url: https://raw.githubusercontent.com/tuProlog/2p-kt/refs/tags/1.0.4/.img/logo.svg
+        sha256: 136d1f65e008d75f36cbdfcacd2c5fa5b3508ad7e29a7b6aea665978b848c944
+        dest-filename: it.unibo.tuprolog.ide.svg
+      - type: file
+        url: https://raw.githubusercontent.com/tuProlog/2p-kt/refs/heads/master/it.unibo.tuprolog.ide.desktop
+        sha256: e90d98bf0d4480035d683692d666d71eaafdaf3c619b20dc2dd20d81ae6f2285
+      - type: file
+        url: https://raw.githubusercontent.com/tuProlog/2p-kt/refs/heads/master/it.unibo.tuprolog.ide.metainfo.xml
```
Comment (bbhtt) [inline it.unibo.tuprolog.ide.yml:41]: urls should be from a tag or a commit otherwise hash will change.

**Comment by VaiTon** [comment]
bot, build it.unibo.tuprolog.ide

**Comment by VaiTon** [comment]
bot, build it.unibo.tuprolog.ide

**Comment by VaiTon** [comment]
bot, build it.unibo.tuprolog.ide

**Comment by VaiTon** [comment]
bot, build it.unibo.tuprolog.ide

**Comment by bbhtt** [comment]
Note that verifying this app will be blocked on https://github.com/flathub-infra/website/issues/3844 `unibo.it` is not in the public suffix list. Is it fine to wait?

Alternatively, if you can upload tokens to `https://unibo.it/.well-known` it will work or you have to use a different app id.

**Comment by gciatto** [comment]
Unibo is a very large university, they usually simply deny any request to affect the main website/domain. I'll try, but it will be hard.

Meanwhile, I guess no verification is bearable... Even if I'm not that familiar with flathub to understand the implications. What are the implications

**Comment by bbhtt** [comment]
The implication is that it will have a yellow "unverified" tick like the other apps you see on flathub.org.

Other than that, it's all the same for us.

**Comment by VaiTon** [comment]
@gciatto 
immagine
vs
immagine

**Comment by gciatto** [comment]
This seems bearable to me :)

Thank you for all these efforts btw

**Comment by VaiTon** [comment]
@gciatto I'm going to make another PR to update the appstream file

**Pull request closed without merging**